// @elastic/eui/lib/components/tour/tour_step_indicator.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "EuiTourStepIndicator", {
  enumerable: true,
  get: function get() {
    return _tour_step_indicator.EuiTourStepIndicator;
  }
});
exports.OuiTourStepIndicator = exports.STATUS = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _common = require("../common");

var _icon = require("../icon");

var _i18n = require("../i18n");

var _tour_step_indicator = require("../../eui_components/tour/tour_step_indicator");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var statusToClassNameMap = {
  complete: 'ouiTourStepIndicator--complete',
  incomplete: 'ouiTourStepIndicator--incomplete',
  active: 'ouiTourStepIndicator--active'
};
var STATUS = (0, _common.keysOf)(statusToClassNameMap);
exports.STATUS = STATUS;

var OuiTourStepIndicator = function OuiTourStepIndicator(_ref) {
  var className = _ref.className,
      number = _ref.number,
      status = _ref.status,
      rest = _objectWithoutProperties(_ref, ["className", "number", "status"]);

  var classes = (0, _classnames.default)('ouiTourStepIndicator', status ? statusToClassNameMap[status] : undefined, className);
  var indicatorIcon;

  if (status === 'active') {
    indicatorIcon = /*#__PURE__*/_react.default.createElement(_i18n.OuiI18n, {
      token: "ouiTourStepIndicator.isActive",
      default: "active"
    }, function (isActive) {
      return /*#__PURE__*/_react.default.createElement(_icon.OuiIcon, {
        type: "dot",
        className: "ouiStepNumber__icon",
        "aria-label": isActive,
        color: "success",
        "aria-current": "step"
      });
    });
  } else if (status === 'complete') {
    indicatorIcon = /*#__PURE__*/_react.default.createElement(_i18n.OuiI18n, {
      token: "ouiTourStepIndicator.isComplete",
      default: "complete"
    }, function (isComplete) {
      return /*#__PURE__*/_react.default.createElement(_icon.OuiIcon, {
        type: "dot",
        className: "ouiStepNumber__icon",
        "aria-label": isComplete,
        color: "subdued"
      });
    });
  } else if (status === 'incomplete') {
    indicatorIcon = /*#__PURE__*/_react.default.createElement(_i18n.OuiI18n, {
      token: "ouiTourStepIndicator.isIncomplete",
      default: "incomplete"
    }, function (isIncomplete) {
      return /*#__PURE__*/_react.default.createElement(_icon.OuiIcon, {
        type: "dot",
        className: "ouiStepNumber__icon",
        "aria-label": isIncomplete,
        color: "subdued"
      });
    });
  }

  return /*#__PURE__*/_react.default.createElement(_i18n.OuiI18n, {
    token: "ouiTourStepIndicator.ariaLabel",
    default: "Step {number} {status}",
    values: {
      status: status,
      number: number
    }
  }, function (ariaLabel) {
    return /*#__PURE__*/_react.default.createElement("li", _extends({
      className: classes,
      "aria-label": ariaLabel
    }, rest), indicatorIcon);
  });
};
/* OUI -> EUI Aliases: Build-Time */


exports.OuiTourStepIndicator = OuiTourStepIndicator;
OuiTourStepIndicator.propTypes = {
  number: _propTypes.default.number.isRequired,
  status: _propTypes.default.oneOf(["complete", "incomplete", "active"]).isRequired,
  className: _propTypes.default.string,
  "aria-label": _propTypes.default.string,
  "data-test-subj": _propTypes.default.string
};
