// @elastic/eui/test-env/components/table/table_footer_cell.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "EuiTableFooterCell", {
  enumerable: true,
  get: function get() {
    return _table_footer_cell.EuiTableFooterCell;
  }
});
exports.OuiTableFooterCell = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _services = require("../../services");

var _utils = require("./utils");

var _table_footer_cell = require("../../eui_components/table/table_footer_cell");

/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
var OuiTableFooterCell = function OuiTableFooterCell(_ref) {
  var children = _ref.children,
      _ref$align = _ref.align,
      align = _ref$align === void 0 ? _services.LEFT_ALIGNMENT : _ref$align,
      className = _ref.className,
      width = _ref.width,
      style = _ref.style,
      rest = (0, _objectWithoutProperties2.default)(_ref, ["children", "align", "className", "width", "style"]);
  var classes = (0, _classnames.default)('ouiTableFooterCell', className);
  var contentClasses = (0, _classnames.default)('ouiTableCellContent', className, {
    'ouiTableCellContent--alignRight': align === _services.RIGHT_ALIGNMENT,
    'ouiTableCellContent--alignCenter': align === _services.CENTER_ALIGNMENT
  });
  var styleObj = (0, _utils.resolveWidthAsStyle)(style, width);
  return /*#__PURE__*/_react.default.createElement("td", (0, _extends2.default)({
    className: classes,
    style: styleObj
  }, rest), /*#__PURE__*/_react.default.createElement("div", {
    className: contentClasses
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "ouiTableCellContent__text"
  }, children)));
};
/* OUI -> EUI Aliases: Build-Time */


exports.OuiTableFooterCell = OuiTableFooterCell;
OuiTableFooterCell.propTypes = {
  className: _propTypes.default.string,
  "aria-label": _propTypes.default.string,
  "data-test-subj": _propTypes.default.string,
  align: _propTypes.default.oneOf(["left", "right", "center"]),
  width: _propTypes.default.oneOfType([_propTypes.default.string.isRequired, _propTypes.default.number.isRequired])
};
