// @elastic/eui/test-env/components/text/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiText", {
  enumerable: true,
  get: function get() {
    return _text.OuiText;
  }
});
Object.defineProperty(exports, "OuiTextColor", {
  enumerable: true,
  get: function get() {
    return _text_color.OuiTextColor;
  }
});
Object.defineProperty(exports, "OuiTextAlign", {
  enumerable: true,
  get: function get() {
    return _text_align.OuiTextAlign;
  }
});
Object.defineProperty(exports, "EuiText", {
  enumerable: true,
  get: function get() {
    return _text2.EuiText;
  }
});
Object.defineProperty(exports, "EuiTextColor", {
  enumerable: true,
  get: function get() {
    return _text_color2.EuiTextColor;
  }
});
Object.defineProperty(exports, "EuiTextAlign", {
  enumerable: true,
  get: function get() {
    return _text_align2.EuiTextAlign;
  }
});

var _text = require("./text");

var _text_color = require("./text_color");

var _text_align = require("./text_align");

var _text2 = require("../../eui_components/text/text");

var _text_color2 = require("../../eui_components/text/text_color");

var _text_align2 = require("../../eui_components/text/text_align");
