// @elastic/eui/test-env/components/flyout/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiFlyout", {
  enumerable: true,
  get: function get() {
    return _flyout.OuiFlyout;
  }
});
Object.defineProperty(exports, "OuiFlyoutBody", {
  enumerable: true,
  get: function get() {
    return _flyout_body.OuiFlyoutBody;
  }
});
Object.defineProperty(exports, "OuiFlyoutFooter", {
  enumerable: true,
  get: function get() {
    return _flyout_footer.OuiFlyoutFooter;
  }
});
Object.defineProperty(exports, "OuiFlyoutHeader", {
  enumerable: true,
  get: function get() {
    return _flyout_header.OuiFlyoutHeader;
  }
});
Object.defineProperty(exports, "EuiFlyout", {
  enumerable: true,
  get: function get() {
    return _flyout2.EuiFlyout;
  }
});
Object.defineProperty(exports, "EuiFlyoutBody", {
  enumerable: true,
  get: function get() {
    return _flyout_body2.EuiFlyoutBody;
  }
});
Object.defineProperty(exports, "EuiFlyoutFooter", {
  enumerable: true,
  get: function get() {
    return _flyout_footer2.EuiFlyoutFooter;
  }
});
Object.defineProperty(exports, "EuiFlyoutHeader", {
  enumerable: true,
  get: function get() {
    return _flyout_header2.EuiFlyoutHeader;
  }
});

var _flyout = require("./flyout");

var _flyout_body = require("./flyout_body");

var _flyout_footer = require("./flyout_footer");

var _flyout_header = require("./flyout_header");

var _flyout2 = require("../../eui_components/flyout/flyout");

var _flyout_body2 = require("../../eui_components/flyout/flyout_body");

var _flyout_footer2 = require("../../eui_components/flyout/flyout_footer");

var _flyout_header2 = require("../../eui_components/flyout/flyout_header");
