// @elastic/eui/lib/components/context_menu/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiContextMenu", {
  enumerable: true,
  get: function get() {
    return _context_menu.OuiContextMenu;
  }
});
Object.defineProperty(exports, "OuiContextMenuPanel", {
  enumerable: true,
  get: function get() {
    return _context_menu_panel.OuiContextMenuPanel;
  }
});
Object.defineProperty(exports, "OuiContextMenuItem", {
  enumerable: true,
  get: function get() {
    return _context_menu_item.OuiContextMenuItem;
  }
});
Object.defineProperty(exports, "EuiContextMenu", {
  enumerable: true,
  get: function get() {
    return _context_menu2.EuiContextMenu;
  }
});
Object.defineProperty(exports, "EuiContextMenuPanel", {
  enumerable: true,
  get: function get() {
    return _context_menu_panel2.EuiContextMenuPanel;
  }
});
Object.defineProperty(exports, "EuiContextMenuItem", {
  enumerable: true,
  get: function get() {
    return _context_menu_item2.EuiContextMenuItem;
  }
});

var _context_menu = require("./context_menu");

var _context_menu_panel = require("./context_menu_panel");

var _context_menu_item = require("./context_menu_item");

var _context_menu2 = require("../../eui_components/context_menu/context_menu");

var _context_menu_panel2 = require("../../eui_components/context_menu/context_menu_panel");

var _context_menu_item2 = require("../../eui_components/context_menu/context_menu_item");
