// @elastic/eui/es/components/loading/index.js
/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
export { OuiLoadingDashboards } from './loading_dashboards';
export { OuiLoadingKibana } from './loading_kibana';
export { OuiLoadingElastic } from './loading_elastic';
export { OuiLoadingChart } from './loading_chart';
export { OuiLoadingContent } from './loading_content';
export { OuiLoadingSpinner } from './loading_spinner';
export { OuiLoadingLogo } from './loading_logo';
/* OUI -> EUI Aliases: Build-Time */

export { EuiLoadingDashboards } from '../../eui_components/loading/loading_dashboards';
/* OUI -> EUI Aliases: Build-Time */

export { EuiLoadingKibana } from '../../eui_components/loading/loading_kibana';
/* OUI -> EUI Aliases: Build-Time */

export { EuiLoadingElastic } from '../../eui_components/loading/loading_elastic';
/* OUI -> EUI Aliases: Build-Time */

export { EuiLoadingChart } from '../../eui_components/loading/loading_chart';
/* OUI -> EUI Aliases: Build-Time */

export { EuiLoadingContent } from '../../eui_components/loading/loading_content';
/* OUI -> EUI Aliases: Build-Time */

export { EuiLoadingSpinner } from '../../eui_components/loading/loading_spinner';
/* OUI -> EUI Aliases: Build-Time */

export { EuiLoadingLogo } from '../../eui_components/loading/loading_logo';
