// @elastic/eui/test-env/components/page/page_header/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiPageHeader", {
  enumerable: true,
  get: function get() {
    return _page_header.OuiPageHeader;
  }
});
Object.defineProperty(exports, "OuiPageHeaderContent", {
  enumerable: true,
  get: function get() {
    return _page_header_content.OuiPageHeaderContent;
  }
});
Object.defineProperty(exports, "OuiPageHeaderSection", {
  enumerable: true,
  get: function get() {
    return _page_header_section.OuiPageHeaderSection;
  }
});
Object.defineProperty(exports, "EuiPageHeader", {
  enumerable: true,
  get: function get() {
    return _page_header2.EuiPageHeader;
  }
});
Object.defineProperty(exports, "EuiPageHeaderContent", {
  enumerable: true,
  get: function get() {
    return _page_header_content2.EuiPageHeaderContent;
  }
});
Object.defineProperty(exports, "EuiPageHeaderSection", {
  enumerable: true,
  get: function get() {
    return _page_header_section2.EuiPageHeaderSection;
  }
});

var _page_header = require("./page_header");

var _page_header_content = require("./page_header_content");

var _page_header_section = require("./page_header_section");

var _page_header2 = require("../../../eui_components/page/page_header/page_header");

var _page_header_content2 = require("../../../eui_components/page/page_header/page_header_content");

var _page_header_section2 = require("../../../eui_components/page/page_header/page_header_section");
