// @elastic/eui/lib/components/comment_list/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiComment", {
  enumerable: true,
  get: function get() {
    return _comment.OuiComment;
  }
});
Object.defineProperty(exports, "OuiCommentEvent", {
  enumerable: true,
  get: function get() {
    return _comment_event.OuiCommentEvent;
  }
});
Object.defineProperty(exports, "OuiCommentTimeline", {
  enumerable: true,
  get: function get() {
    return _comment_timeline.OuiCommentTimeline;
  }
});
Object.defineProperty(exports, "OuiCommentList", {
  enumerable: true,
  get: function get() {
    return _comment_list.OuiCommentList;
  }
});
Object.defineProperty(exports, "EuiComment", {
  enumerable: true,
  get: function get() {
    return _comment2.EuiComment;
  }
});
Object.defineProperty(exports, "EuiCommentEvent", {
  enumerable: true,
  get: function get() {
    return _comment_event2.EuiCommentEvent;
  }
});
Object.defineProperty(exports, "EuiCommentTimeline", {
  enumerable: true,
  get: function get() {
    return _comment_timeline2.EuiCommentTimeline;
  }
});
Object.defineProperty(exports, "EuiCommentList", {
  enumerable: true,
  get: function get() {
    return _comment_list2.EuiCommentList;
  }
});

var _comment = require("./comment");

var _comment_event = require("./comment_event");

var _comment_timeline = require("./comment_timeline");

var _comment_list = require("./comment_list");

var _comment2 = require("../../eui_components/comment_list/comment");

var _comment_event2 = require("../../eui_components/comment_list/comment_event");

var _comment_timeline2 = require("../../eui_components/comment_list/comment_timeline");

var _comment_list2 = require("../../eui_components/comment_list/comment_list");
