// @elastic/eui/test-env/components/accessibility/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiKeyboardAccessible", {
  enumerable: true,
  get: function get() {
    return _keyboard_accessible.OuiKeyboardAccessible;
  }
});
Object.defineProperty(exports, "OuiScreenReaderOnly", {
  enumerable: true,
  get: function get() {
    return _screen_reader.OuiScreenReaderOnly;
  }
});
Object.defineProperty(exports, "OuiSkipLink", {
  enumerable: true,
  get: function get() {
    return _skip_link.OuiSkipLink;
  }
});
Object.defineProperty(exports, "EuiKeyboardAccessible", {
  enumerable: true,
  get: function get() {
    return _keyboard_accessible2.EuiKeyboardAccessible;
  }
});
Object.defineProperty(exports, "EuiScreenReaderOnly", {
  enumerable: true,
  get: function get() {
    return _screen_reader2.EuiScreenReaderOnly;
  }
});
Object.defineProperty(exports, "EuiSkipLink", {
  enumerable: true,
  get: function get() {
    return _skip_link2.EuiSkipLink;
  }
});

var _keyboard_accessible = require("./keyboard_accessible");

var _screen_reader = require("./screen_reader");

var _skip_link = require("./skip_link");

var _keyboard_accessible2 = require("../../eui_components/accessibility/keyboard_accessible");

var _screen_reader2 = require("../../eui_components/accessibility/screen_reader");

var _skip_link2 = require("../../eui_components/accessibility/skip_link");
