// @elastic/eui/test-env/components/loading/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiLoadingDashboards", {
  enumerable: true,
  get: function get() {
    return _loading_dashboards.OuiLoadingDashboards;
  }
});
Object.defineProperty(exports, "OuiLoadingKibana", {
  enumerable: true,
  get: function get() {
    return _loading_kibana.OuiLoadingKibana;
  }
});
Object.defineProperty(exports, "OuiLoadingElastic", {
  enumerable: true,
  get: function get() {
    return _loading_elastic.OuiLoadingElastic;
  }
});
Object.defineProperty(exports, "OuiLoadingChart", {
  enumerable: true,
  get: function get() {
    return _loading_chart.OuiLoadingChart;
  }
});
Object.defineProperty(exports, "OuiLoadingContent", {
  enumerable: true,
  get: function get() {
    return _loading_content.OuiLoadingContent;
  }
});
Object.defineProperty(exports, "OuiLoadingSpinner", {
  enumerable: true,
  get: function get() {
    return _loading_spinner.OuiLoadingSpinner;
  }
});
Object.defineProperty(exports, "OuiLoadingLogo", {
  enumerable: true,
  get: function get() {
    return _loading_logo.OuiLoadingLogo;
  }
});
Object.defineProperty(exports, "EuiLoadingDashboards", {
  enumerable: true,
  get: function get() {
    return _loading_dashboards2.EuiLoadingDashboards;
  }
});
Object.defineProperty(exports, "EuiLoadingKibana", {
  enumerable: true,
  get: function get() {
    return _loading_kibana2.EuiLoadingKibana;
  }
});
Object.defineProperty(exports, "EuiLoadingElastic", {
  enumerable: true,
  get: function get() {
    return _loading_elastic2.EuiLoadingElastic;
  }
});
Object.defineProperty(exports, "EuiLoadingChart", {
  enumerable: true,
  get: function get() {
    return _loading_chart2.EuiLoadingChart;
  }
});
Object.defineProperty(exports, "EuiLoadingContent", {
  enumerable: true,
  get: function get() {
    return _loading_content2.EuiLoadingContent;
  }
});
Object.defineProperty(exports, "EuiLoadingSpinner", {
  enumerable: true,
  get: function get() {
    return _loading_spinner2.EuiLoadingSpinner;
  }
});
Object.defineProperty(exports, "EuiLoadingLogo", {
  enumerable: true,
  get: function get() {
    return _loading_logo2.EuiLoadingLogo;
  }
});

var _loading_dashboards = require("./loading_dashboards");

var _loading_kibana = require("./loading_kibana");

var _loading_elastic = require("./loading_elastic");

var _loading_chart = require("./loading_chart");

var _loading_content = require("./loading_content");

var _loading_spinner = require("./loading_spinner");

var _loading_logo = require("./loading_logo");

var _loading_dashboards2 = require("../../eui_components/loading/loading_dashboards");

var _loading_kibana2 = require("../../eui_components/loading/loading_kibana");

var _loading_elastic2 = require("../../eui_components/loading/loading_elastic");

var _loading_chart2 = require("../../eui_components/loading/loading_chart");

var _loading_content2 = require("../../eui_components/loading/loading_content");

var _loading_spinner2 = require("../../eui_components/loading/loading_spinner");

var _loading_logo2 = require("../../eui_components/loading/loading_logo");
