// @elastic/eui/lib/components/page/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiPage", {
  enumerable: true,
  get: function get() {
    return _page.OuiPage;
  }
});
Object.defineProperty(exports, "OuiPageBody", {
  enumerable: true,
  get: function get() {
    return _page_body.OuiPageBody;
  }
});
Object.defineProperty(exports, "OuiPageContent", {
  enumerable: true,
  get: function get() {
    return _page_content.OuiPageContent;
  }
});
Object.defineProperty(exports, "OuiPageContentBody", {
  enumerable: true,
  get: function get() {
    return _page_content.OuiPageContentBody;
  }
});
Object.defineProperty(exports, "OuiPageContentHeader", {
  enumerable: true,
  get: function get() {
    return _page_content.OuiPageContentHeader;
  }
});
Object.defineProperty(exports, "OuiPageContentHeaderSection", {
  enumerable: true,
  get: function get() {
    return _page_content.OuiPageContentHeaderSection;
  }
});
Object.defineProperty(exports, "OuiPageHeader", {
  enumerable: true,
  get: function get() {
    return _page_header.OuiPageHeader;
  }
});
Object.defineProperty(exports, "OuiPageHeaderContent", {
  enumerable: true,
  get: function get() {
    return _page_header.OuiPageHeaderContent;
  }
});
Object.defineProperty(exports, "OuiPageHeaderSection", {
  enumerable: true,
  get: function get() {
    return _page_header.OuiPageHeaderSection;
  }
});
Object.defineProperty(exports, "OuiPageSideBar", {
  enumerable: true,
  get: function get() {
    return _page_side_bar.OuiPageSideBar;
  }
});
Object.defineProperty(exports, "OuiPageTemplate", {
  enumerable: true,
  get: function get() {
    return _page_template.OuiPageTemplate;
  }
});
Object.defineProperty(exports, "EuiPage", {
  enumerable: true,
  get: function get() {
    return _page2.EuiPage;
  }
});
Object.defineProperty(exports, "EuiPageBody", {
  enumerable: true,
  get: function get() {
    return _page_body2.EuiPageBody;
  }
});
Object.defineProperty(exports, "EuiPageContent", {
  enumerable: true,
  get: function get() {
    return _page_content2.EuiPageContent;
  }
});
Object.defineProperty(exports, "EuiPageContentBody", {
  enumerable: true,
  get: function get() {
    return _page_content2.EuiPageContentBody;
  }
});
Object.defineProperty(exports, "EuiPageContentHeader", {
  enumerable: true,
  get: function get() {
    return _page_content2.EuiPageContentHeader;
  }
});
Object.defineProperty(exports, "EuiPageContentHeaderSection", {
  enumerable: true,
  get: function get() {
    return _page_content2.EuiPageContentHeaderSection;
  }
});
Object.defineProperty(exports, "EuiPageHeader", {
  enumerable: true,
  get: function get() {
    return _page_header2.EuiPageHeader;
  }
});
Object.defineProperty(exports, "EuiPageHeaderContent", {
  enumerable: true,
  get: function get() {
    return _page_header2.EuiPageHeaderContent;
  }
});
Object.defineProperty(exports, "EuiPageHeaderSection", {
  enumerable: true,
  get: function get() {
    return _page_header2.EuiPageHeaderSection;
  }
});
Object.defineProperty(exports, "EuiPageSideBar", {
  enumerable: true,
  get: function get() {
    return _page_side_bar2.EuiPageSideBar;
  }
});
Object.defineProperty(exports, "EuiPageTemplate", {
  enumerable: true,
  get: function get() {
    return _page_template2.EuiPageTemplate;
  }
});

var _page = require("./page");

var _page_body = require("./page_body");

var _page_content = require("./page_content");

var _page_header = require("./page_header");

var _page_side_bar = require("./page_side_bar");

var _page_template = require("./page_template");

var _page2 = require("../../eui_components/page/page");

var _page_body2 = require("../../eui_components/page/page_body");

var _page_content2 = require("../../eui_components/page/page_content");

var _page_header2 = require("../../eui_components/page/page_header");

var _page_side_bar2 = require("../../eui_components/page/page_side_bar");

var _page_template2 = require("../../eui_components/page/page_template");
