// @elastic/eui/test-env/components/description_list/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiDescriptionList", {
  enumerable: true,
  get: function get() {
    return _description_list.OuiDescriptionList;
  }
});
Object.defineProperty(exports, "OuiDescriptionListTitle", {
  enumerable: true,
  get: function get() {
    return _description_list_title.OuiDescriptionListTitle;
  }
});
Object.defineProperty(exports, "OuiDescriptionListDescription", {
  enumerable: true,
  get: function get() {
    return _description_list_description.OuiDescriptionListDescription;
  }
});
Object.defineProperty(exports, "EuiDescriptionList", {
  enumerable: true,
  get: function get() {
    return _description_list2.EuiDescriptionList;
  }
});
Object.defineProperty(exports, "EuiDescriptionListTitle", {
  enumerable: true,
  get: function get() {
    return _description_list_title2.EuiDescriptionListTitle;
  }
});
Object.defineProperty(exports, "EuiDescriptionListDescription", {
  enumerable: true,
  get: function get() {
    return _description_list_description2.EuiDescriptionListDescription;
  }
});

var _description_list = require("./description_list");

var _description_list_title = require("./description_list_title");

var _description_list_description = require("./description_list_description");

var _description_list2 = require("../../eui_components/description_list/description_list");

var _description_list_title2 = require("../../eui_components/description_list/description_list_title");

var _description_list_description2 = require("../../eui_components/description_list/description_list_description");
