// @elastic/eui/lib/components/color_picker/color_palette_display/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiColorPaletteDisplay", {
  enumerable: true,
  get: function get() {
    return _color_palette_display.OuiColorPaletteDisplay;
  }
});
Object.defineProperty(exports, "OuiColorPaletteDisplayFixed", {
  enumerable: true,
  get: function get() {
    return _color_palette_display_fixed.OuiColorPaletteDisplayFixed;
  }
});
Object.defineProperty(exports, "OuiColorPaletteDisplayGradient", {
  enumerable: true,
  get: function get() {
    return _color_palette_display_gradient.OuiColorPaletteDisplayGradient;
  }
});
Object.defineProperty(exports, "EuiColorPaletteDisplay", {
  enumerable: true,
  get: function get() {
    return _color_palette_display2.EuiColorPaletteDisplay;
  }
});
Object.defineProperty(exports, "EuiColorPaletteDisplayFixed", {
  enumerable: true,
  get: function get() {
    return _color_palette_display_fixed2.EuiColorPaletteDisplayFixed;
  }
});
Object.defineProperty(exports, "EuiColorPaletteDisplayGradient", {
  enumerable: true,
  get: function get() {
    return _color_palette_display_gradient2.EuiColorPaletteDisplayGradient;
  }
});

var _color_palette_display = require("./color_palette_display");

var _color_palette_display_fixed = require("./color_palette_display_fixed");

var _color_palette_display_gradient = require("./color_palette_display_gradient");

var _color_palette_display2 = require("../../../eui_components/color_picker/color_palette_display/color_palette_display");

var _color_palette_display_fixed2 = require("../../../eui_components/color_picker/color_palette_display/color_palette_display_fixed");

var _color_palette_display_gradient2 = require("../../../eui_components/color_picker/color_palette_display/color_palette_display_gradient");
