// @elastic/eui/lib/components/badge/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiBadge", {
  enumerable: true,
  get: function get() {
    return _badge.OuiBadge;
  }
});
Object.defineProperty(exports, "OuiBetaBadge", {
  enumerable: true,
  get: function get() {
    return _beta_badge.OuiBetaBadge;
  }
});
Object.defineProperty(exports, "OuiNotificationBadge", {
  enumerable: true,
  get: function get() {
    return _notification_badge.OuiNotificationBadge;
  }
});
Object.defineProperty(exports, "OuiBadgeGroup", {
  enumerable: true,
  get: function get() {
    return _badge_group.OuiBadgeGroup;
  }
});
Object.defineProperty(exports, "EuiBadge", {
  enumerable: true,
  get: function get() {
    return _badge2.EuiBadge;
  }
});
Object.defineProperty(exports, "EuiBetaBadge", {
  enumerable: true,
  get: function get() {
    return _beta_badge2.EuiBetaBadge;
  }
});
Object.defineProperty(exports, "EuiNotificationBadge", {
  enumerable: true,
  get: function get() {
    return _notification_badge2.EuiNotificationBadge;
  }
});
Object.defineProperty(exports, "EuiBadgeGroup", {
  enumerable: true,
  get: function get() {
    return _badge_group2.EuiBadgeGroup;
  }
});

var _badge = require("./badge");

var _beta_badge = require("./beta_badge");

var _notification_badge = require("./notification_badge");

var _badge_group = require("./badge_group");

var _badge2 = require("../../eui_components/badge/badge");

var _beta_badge2 = require("../../eui_components/badge/beta_badge");

var _notification_badge2 = require("../../eui_components/badge/notification_badge");

var _badge_group2 = require("../../eui_components/badge/badge_group");
