// @elastic/eui/test-env/components/table/mobile/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiTableHeaderMobile", {
  enumerable: true,
  get: function get() {
    return _table_header_mobile.OuiTableHeaderMobile;
  }
});
Object.defineProperty(exports, "OuiTableSortMobile", {
  enumerable: true,
  get: function get() {
    return _table_sort_mobile.OuiTableSortMobile;
  }
});
Object.defineProperty(exports, "OuiTableSortMobileItem", {
  enumerable: true,
  get: function get() {
    return _table_sort_mobile_item.OuiTableSortMobileItem;
  }
});
Object.defineProperty(exports, "EuiTableHeaderMobile", {
  enumerable: true,
  get: function get() {
    return _table_header_mobile2.EuiTableHeaderMobile;
  }
});
Object.defineProperty(exports, "EuiTableSortMobile", {
  enumerable: true,
  get: function get() {
    return _table_sort_mobile2.EuiTableSortMobile;
  }
});
Object.defineProperty(exports, "EuiTableSortMobileItem", {
  enumerable: true,
  get: function get() {
    return _table_sort_mobile_item2.EuiTableSortMobileItem;
  }
});

var _table_header_mobile = require("./table_header_mobile");

var _table_sort_mobile = require("./table_sort_mobile");

var _table_sort_mobile_item = require("./table_sort_mobile_item");

var _table_header_mobile2 = require("../../../eui_components/table/mobile/table_header_mobile");

var _table_sort_mobile2 = require("../../../eui_components/table/mobile/table_sort_mobile");

var _table_sort_mobile_item2 = require("../../../eui_components/table/mobile/table_sort_mobile_item");
