// @elastic/eui/test-env/components/flex/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiFlexGroup", {
  enumerable: true,
  get: function get() {
    return _flex_group.OuiFlexGroup;
  }
});
Object.defineProperty(exports, "OuiFlexGrid", {
  enumerable: true,
  get: function get() {
    return _flex_grid.OuiFlexGrid;
  }
});
Object.defineProperty(exports, "OuiFlexItem", {
  enumerable: true,
  get: function get() {
    return _flex_item.OuiFlexItem;
  }
});
Object.defineProperty(exports, "EuiFlexGroup", {
  enumerable: true,
  get: function get() {
    return _flex_group2.EuiFlexGroup;
  }
});
Object.defineProperty(exports, "EuiFlexGrid", {
  enumerable: true,
  get: function get() {
    return _flex_grid2.EuiFlexGrid;
  }
});
Object.defineProperty(exports, "EuiFlexItem", {
  enumerable: true,
  get: function get() {
    return _flex_item2.EuiFlexItem;
  }
});

var _flex_group = require("./flex_group");

var _flex_grid = require("./flex_grid");

var _flex_item = require("./flex_item");

var _flex_group2 = require("../../eui_components/flex/flex_group");

var _flex_grid2 = require("../../eui_components/flex/flex_grid");

var _flex_item2 = require("../../eui_components/flex/flex_item");
