// @elastic/eui/test-env/components/modal/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiConfirmModal", {
  enumerable: true,
  get: function get() {
    return _confirm_modal.OuiConfirmModal;
  }
});
Object.defineProperty(exports, "OUI_MODAL_CONFIRM_BUTTON", {
  enumerable: true,
  get: function get() {
    return _confirm_modal.CONFIRM_BUTTON;
  }
});
Object.defineProperty(exports, "OUI_MODAL_CANCEL_BUTTON", {
  enumerable: true,
  get: function get() {
    return _confirm_modal.CANCEL_BUTTON;
  }
});
Object.defineProperty(exports, "OuiModal", {
  enumerable: true,
  get: function get() {
    return _modal.OuiModal;
  }
});
Object.defineProperty(exports, "OuiModalFooter", {
  enumerable: true,
  get: function get() {
    return _modal_footer.OuiModalFooter;
  }
});
Object.defineProperty(exports, "OuiModalHeader", {
  enumerable: true,
  get: function get() {
    return _modal_header.OuiModalHeader;
  }
});
Object.defineProperty(exports, "OuiModalBody", {
  enumerable: true,
  get: function get() {
    return _modal_body.OuiModalBody;
  }
});
Object.defineProperty(exports, "OuiModalHeaderTitle", {
  enumerable: true,
  get: function get() {
    return _modal_header_title.OuiModalHeaderTitle;
  }
});
Object.defineProperty(exports, "EuiConfirmModal", {
  enumerable: true,
  get: function get() {
    return _confirm_modal2.EuiConfirmModal;
  }
});
Object.defineProperty(exports, "EUI_MODAL_CONFIRM_BUTTON", {
  enumerable: true,
  get: function get() {
    return _confirm_modal2.CONFIRM_BUTTON;
  }
});
Object.defineProperty(exports, "EUI_MODAL_CANCEL_BUTTON", {
  enumerable: true,
  get: function get() {
    return _confirm_modal2.CANCEL_BUTTON;
  }
});
Object.defineProperty(exports, "EuiModal", {
  enumerable: true,
  get: function get() {
    return _modal2.EuiModal;
  }
});
Object.defineProperty(exports, "EuiModalFooter", {
  enumerable: true,
  get: function get() {
    return _modal_footer2.EuiModalFooter;
  }
});
Object.defineProperty(exports, "EuiModalHeader", {
  enumerable: true,
  get: function get() {
    return _modal_header2.EuiModalHeader;
  }
});
Object.defineProperty(exports, "EuiModalBody", {
  enumerable: true,
  get: function get() {
    return _modal_body2.EuiModalBody;
  }
});
Object.defineProperty(exports, "EuiModalHeaderTitle", {
  enumerable: true,
  get: function get() {
    return _modal_header_title2.EuiModalHeaderTitle;
  }
});

var _confirm_modal = require("./confirm_modal");

var _modal = require("./modal");

var _modal_footer = require("./modal_footer");

var _modal_header = require("./modal_header");

var _modal_body = require("./modal_body");

var _modal_header_title = require("./modal_header_title");

var _confirm_modal2 = require("../../eui_components/modal/confirm_modal");

var _modal2 = require("../../eui_components/modal/modal");

var _modal_footer2 = require("../../eui_components/modal/modal_footer");

var _modal_header2 = require("../../eui_components/modal/modal_header");

var _modal_body2 = require("../../eui_components/modal/modal_body");

var _modal_header_title2 = require("../../eui_components/modal/modal_header_title");
