// @elastic/eui/test-env/components/selectable/selectable_templates/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiSelectableTemplateSitewide", {
  enumerable: true,
  get: function get() {
    return _selectable_template_sitewide.OuiSelectableTemplateSitewide;
  }
});
Object.defineProperty(exports, "ouiSelectableTemplateSitewideFormatOptions", {
  enumerable: true,
  get: function get() {
    return _selectable_template_sitewide_option.ouiSelectableTemplateSitewideFormatOptions;
  }
});
Object.defineProperty(exports, "ouiSelectableTemplateSitewideRenderOptions", {
  enumerable: true,
  get: function get() {
    return _selectable_template_sitewide_option.ouiSelectableTemplateSitewideRenderOptions;
  }
});
Object.defineProperty(exports, "EuiSelectableTemplateSitewide", {
  enumerable: true,
  get: function get() {
    return _selectable_template_sitewide2.EuiSelectableTemplateSitewide;
  }
});
Object.defineProperty(exports, "euiSelectableTemplateSitewideFormatOptions", {
  enumerable: true,
  get: function get() {
    return _selectable_template_sitewide_option2.euiSelectableTemplateSitewideFormatOptions;
  }
});
Object.defineProperty(exports, "euiSelectableTemplateSitewideRenderOptions", {
  enumerable: true,
  get: function get() {
    return _selectable_template_sitewide_option2.euiSelectableTemplateSitewideRenderOptions;
  }
});

var _selectable_template_sitewide = require("./selectable_template_sitewide");

var _selectable_template_sitewide_option = require("./selectable_template_sitewide_option");

var _selectable_template_sitewide2 = require("../../../eui_components/selectable/selectable_templates/selectable_template_sitewide");

var _selectable_template_sitewide_option2 = require("../../../eui_components/selectable/selectable_templates/selectable_template_sitewide_option");
