// @elastic/eui/lib/components/suggest/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiSuggestInput", {
  enumerable: true,
  get: function get() {
    return _suggest_input.OuiSuggestInput;
  }
});
Object.defineProperty(exports, "OuiSuggestItem", {
  enumerable: true,
  get: function get() {
    return _suggest_item.OuiSuggestItem;
  }
});
Object.defineProperty(exports, "OuiSuggest", {
  enumerable: true,
  get: function get() {
    return _suggest.OuiSuggest;
  }
});
Object.defineProperty(exports, "EuiSuggestInput", {
  enumerable: true,
  get: function get() {
    return _suggest_input2.EuiSuggestInput;
  }
});
Object.defineProperty(exports, "EuiSuggestItem", {
  enumerable: true,
  get: function get() {
    return _suggest_item2.EuiSuggestItem;
  }
});
Object.defineProperty(exports, "EuiSuggest", {
  enumerable: true,
  get: function get() {
    return _suggest2.EuiSuggest;
  }
});

var _suggest_input = require("./suggest_input");

var _suggest_item = require("./suggest_item");

var _suggest = require("./suggest");

var _suggest_input2 = require("../../eui_components/suggest/suggest_input");

var _suggest_item2 = require("../../eui_components/suggest/suggest_item");

var _suggest2 = require("../../eui_components/suggest/suggest");
