// @elastic/eui/test-env/components/code/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiCode", {
  enumerable: true,
  get: function get() {
    return _code.OuiCode;
  }
});
Object.defineProperty(exports, "OuiCodeBlock", {
  enumerable: true,
  get: function get() {
    return _code_block.OuiCodeBlock;
  }
});
Object.defineProperty(exports, "OuiCodeBlockImpl", {
  enumerable: true,
  get: function get() {
    return _code_block2.OuiCodeBlockImpl;
  }
});
Object.defineProperty(exports, "EuiCode", {
  enumerable: true,
  get: function get() {
    return _code2.EuiCode;
  }
});
Object.defineProperty(exports, "EuiCodeBlock", {
  enumerable: true,
  get: function get() {
    return _code_block3.EuiCodeBlock;
  }
});
Object.defineProperty(exports, "EuiCodeBlockImpl", {
  enumerable: true,
  get: function get() {
    return _code_block4.EuiCodeBlockImpl;
  }
});

var _code = require("./code");

var _code_block = require("./code_block");

var _code_block2 = require("./_code_block");

var _code2 = require("../../eui_components/code/code");

var _code_block3 = require("../../eui_components/code/code_block");

var _code_block4 = require("../../eui_components/code/_code_block");
