// @elastic/eui/test-env/components/page/page_content/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiPageContent", {
  enumerable: true,
  get: function get() {
    return _page_content.OuiPageContent;
  }
});
Object.defineProperty(exports, "OuiPageContentBody", {
  enumerable: true,
  get: function get() {
    return _page_content_body.OuiPageContentBody;
  }
});
Object.defineProperty(exports, "OuiPageContentHeader", {
  enumerable: true,
  get: function get() {
    return _page_content_header.OuiPageContentHeader;
  }
});
Object.defineProperty(exports, "OuiPageContentHeaderSection", {
  enumerable: true,
  get: function get() {
    return _page_content_header_section.OuiPageContentHeaderSection;
  }
});
Object.defineProperty(exports, "EuiPageContent", {
  enumerable: true,
  get: function get() {
    return _page_content2.EuiPageContent;
  }
});
Object.defineProperty(exports, "EuiPageContentBody", {
  enumerable: true,
  get: function get() {
    return _page_content_body2.EuiPageContentBody;
  }
});
Object.defineProperty(exports, "EuiPageContentHeader", {
  enumerable: true,
  get: function get() {
    return _page_content_header2.EuiPageContentHeader;
  }
});
Object.defineProperty(exports, "EuiPageContentHeaderSection", {
  enumerable: true,
  get: function get() {
    return _page_content_header_section2.EuiPageContentHeaderSection;
  }
});

var _page_content = require("./page_content");

var _page_content_body = require("./page_content_body");

var _page_content_header = require("./page_content_header");

var _page_content_header_section = require("./page_content_header_section");

var _page_content2 = require("../../../eui_components/page/page_content/page_content");

var _page_content_body2 = require("../../../eui_components/page/page_content/page_content_body");

var _page_content_header2 = require("../../../eui_components/page/page_content/page_content_header");

var _page_content_header_section2 = require("../../../eui_components/page/page_content/page_content_header_section");
