// @elastic/eui/es/components/accessibility/index.js
/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
export { OuiKeyboardAccessible } from './keyboard_accessible';
export { OuiScreenReaderOnly } from './screen_reader';
export { OuiSkipLink } from './skip_link';
/* OUI -> EUI Aliases: Build-Time */

export { EuiKeyboardAccessible } from '../../eui_components/accessibility/keyboard_accessible';
/* OUI -> EUI Aliases: Build-Time */

export { EuiScreenReaderOnly } from '../../eui_components/accessibility/screen_reader';
/* OUI -> EUI Aliases: Build-Time */

export { EuiSkipLink } from '../../eui_components/accessibility/skip_link';
