// @elastic/eui/test-env/components/header/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiHeader", {
  enumerable: true,
  get: function get() {
    return _header.OuiHeader;
  }
});
Object.defineProperty(exports, "OuiHeaderAlert", {
  enumerable: true,
  get: function get() {
    return _header_alert.OuiHeaderAlert;
  }
});
Object.defineProperty(exports, "OuiHeaderBreadcrumbs", {
  enumerable: true,
  get: function get() {
    return _header_breadcrumbs.OuiHeaderBreadcrumbs;
  }
});
Object.defineProperty(exports, "OuiHeaderLink", {
  enumerable: true,
  get: function get() {
    return _header_links.OuiHeaderLink;
  }
});
Object.defineProperty(exports, "OuiHeaderLinks", {
  enumerable: true,
  get: function get() {
    return _header_links.OuiHeaderLinks;
  }
});
Object.defineProperty(exports, "OuiHeaderLogo", {
  enumerable: true,
  get: function get() {
    return _header_logo.OuiHeaderLogo;
  }
});
Object.defineProperty(exports, "OuiHeaderSection", {
  enumerable: true,
  get: function get() {
    return _header_section.OuiHeaderSection;
  }
});
Object.defineProperty(exports, "OuiHeaderSectionItem", {
  enumerable: true,
  get: function get() {
    return _header_section.OuiHeaderSectionItem;
  }
});
Object.defineProperty(exports, "OuiHeaderSectionItemButton", {
  enumerable: true,
  get: function get() {
    return _header_section.OuiHeaderSectionItemButton;
  }
});
Object.defineProperty(exports, "EuiHeader", {
  enumerable: true,
  get: function get() {
    return _header2.EuiHeader;
  }
});
Object.defineProperty(exports, "EuiHeaderAlert", {
  enumerable: true,
  get: function get() {
    return _header_alert2.EuiHeaderAlert;
  }
});
Object.defineProperty(exports, "EuiHeaderBreadcrumbs", {
  enumerable: true,
  get: function get() {
    return _header_breadcrumbs2.EuiHeaderBreadcrumbs;
  }
});
Object.defineProperty(exports, "EuiHeaderLink", {
  enumerable: true,
  get: function get() {
    return _header_links2.EuiHeaderLink;
  }
});
Object.defineProperty(exports, "EuiHeaderLinks", {
  enumerable: true,
  get: function get() {
    return _header_links2.EuiHeaderLinks;
  }
});
Object.defineProperty(exports, "EuiHeaderLogo", {
  enumerable: true,
  get: function get() {
    return _header_logo2.EuiHeaderLogo;
  }
});
Object.defineProperty(exports, "EuiHeaderSection", {
  enumerable: true,
  get: function get() {
    return _header_section2.EuiHeaderSection;
  }
});
Object.defineProperty(exports, "EuiHeaderSectionItem", {
  enumerable: true,
  get: function get() {
    return _header_section2.EuiHeaderSectionItem;
  }
});
Object.defineProperty(exports, "EuiHeaderSectionItemButton", {
  enumerable: true,
  get: function get() {
    return _header_section2.EuiHeaderSectionItemButton;
  }
});

var _header = require("./header");

var _header_alert = require("./header_alert");

var _header_breadcrumbs = require("./header_breadcrumbs");

var _header_links = require("./header_links");

var _header_logo = require("./header_logo");

var _header_section = require("./header_section");

var _header2 = require("../../eui_components/header/header");

var _header_alert2 = require("../../eui_components/header/header_alert");

var _header_breadcrumbs2 = require("../../eui_components/header/header_breadcrumbs");

var _header_links2 = require("../../eui_components/header/header_links");

var _header_logo2 = require("../../eui_components/header/header_logo");

var _header_section2 = require("../../eui_components/header/header_section");
