// @elastic/eui/lib/components/tabs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OuiTab", {
  enumerable: true,
  get: function get() {
    return _tab.OuiTab;
  }
});
Object.defineProperty(exports, "OuiTabs", {
  enumerable: true,
  get: function get() {
    return _tabs.OuiTabs;
  }
});
Object.defineProperty(exports, "OuiTabbedContent", {
  enumerable: true,
  get: function get() {
    return _tabbed_content.OuiTabbedContent;
  }
});
Object.defineProperty(exports, "EuiTab", {
  enumerable: true,
  get: function get() {
    return _tab2.EuiTab;
  }
});
Object.defineProperty(exports, "EuiTabs", {
  enumerable: true,
  get: function get() {
    return _tabs2.EuiTabs;
  }
});
Object.defineProperty(exports, "EuiTabbedContent", {
  enumerable: true,
  get: function get() {
    return _tabbed_content2.EuiTabbedContent;
  }
});

var _tab = require("./tab");

var _tabs = require("./tabs");

var _tabbed_content = require("./tabbed_content");

var _tab2 = require("../../eui_components/tabs/tab");

var _tabs2 = require("../../eui_components/tabs/tabs");

var _tabbed_content2 = require("../../eui_components/tabs/tabbed_content");
